Add missing headers to azure-attestation\README.md

mwynne opened the pull request:
# Description

Added sections per #4554. This is a work in progress. 

 @scbedd before I get really invested with the content under the new headers, I just wanted to make sure I'm going in the right direction with this. It's a pretty old issue, and I noticed that a few of the links in the original issue are broken and it appears there has been some refactoring in the repo since. Thanks.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

mwynne commented:
@microsoft-github-policy-service agree

scbedd commented on sdk/attestation/azure-mgmt-attestation/README.md:
```diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
 Code samples for this package can be found at [Attestation Management](https://docs.microsoft.com/samples/browse/?languages=python&term=Getting%20started%20-%20Managing&terms=Getting%20started%20-%20Managing) on docs.microsoft.com.
 Additional code samples for different Azure services are available at [Samples Repo](https://aka.ms/azsdk/python/mgmt/samples)
 
+## Key Concepts
```
```suggestion
```

scbedd commented on sdk/attestation/azure-mgmt-attestation/README.md:
```diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
 Additional code samples for different Azure services are available at [Samples Repo](https://aka.ms/azsdk/python/mgmt/samples)
 
+## Key Concepts
+Content TBD
```
```suggestion
```

scbedd commented on sdk/attestation/azure-mgmt-attestation/README.md:
````diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
+response = client.attestation_providers.list()
+print(response)
+```
+TBC
````
```suggestion
```

scbedd commented on sdk/attestation/azure-mgmt-attestation/README.md:
```diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
 
-# Provide Feedback
+### Permission issues
+Service client calls that result in HttpResponseError with a StatusCode of 401 or 403 often indicate the caller doesn't have sufficient permissions for the specified API. Check the service documentation to determine which RBAC roles are needed for the specific request, and ensure the authenticated user or service principal have been granted the appropriate roles on the resource.
```
```suggestion
Service client calls that result in HttpResponseError with a StatusCode of 401 or 403 often indicate the caller doesn't have sufficient permissions for the specified API. Check the [service documentation](https://learn.microsoft.com/en-us/azure/attestation/) to determine which RBAC roles are needed for the specific request, and ensure the authenticated user or service principal have been granted the appropriate roles on the resource.
```

scbedd reviewed (changes requested):
Few nits cleaning up TBC/TBD and adding a link to the service documentation.

mwynne commented:
@scbedd I'm glad I checked before moving ahead with anything else! I addressed your comments in a new commit. Let me know if I missed anything, or if you'd like additional changes. 

I'll go find something else to work on. Since I'm new around here, if there are any issues that no one else wants to volunteer for that you may want to suggest, please let me know.

scbedd reviewed (approved):
Thank you for the contribution @mwynne ! I'm going to ignore this error (which has nothing to do with your PR) and get this merged.

scbedd commented:
/check-enforcer override

Overriding as the `mypy` error is most definitively not from this PR.
